### PR TITLE
[IRGen] Correctly emit metadata for ObjC class properties

### PR DESCRIPTION
**Pull request by jrose-apple**
...which is important because Clang CodeGen was already claiming we were!

Follow-up to 50e3b337. Resolves rdar://problem/25473573.

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by jrose-apple** [comment]
@gparker42, mind reviewing?

**Comment by jrose-apple** [comment]
@swift-ci Please test

File: lib/IRGen/GenClass.cpp [hunk 1]
```diff
@@ -918,62 +919,68 @@ namespace {
...
-      fields.push_back(buildClassMethodList());
+      fields[3] = buildClassMethodList();
       //   const protocol_list_t *baseProtocols;
-      fields.push_back(buildProtocolList());
+      fields[4] = buildProtocolList();
       //   const property_list_t *properties;
-      fields.push_back(buildPropertyList());
+      fields[5] = buildPropertyList(ForClass);
+      //   const property_list_t *classProperties;
+      fields[6] = buildPropertyList(ForMetaClass);
```
Comment (gparker42) [inline lib/IRGen/GenClass.cpp]: We added a `uint32_t size = sizeof(category_t)` field here (rdar://24804226).

**Comment by gparker42** [comment]
Where are the objc image info flags set? There is a new one to indicate the presence of this data.

Looks good other than the image info flags and the new category_t.size field.

**Comment by jrose-apple** [comment]
Clang is setting them for us, se we've just been broken for the last few weeks. Oops.

Is there a good way to test that that doesn't depend on updates to the ObjC runtime?

**Comment by gparker42** [comment]
You can write a runtime test for class properties on class objects, at least, because the old runtime happens to handle them correctly. Check the result of `class_getProperty(object_getClass([YourClass class]), "Your PropertyName")`.

**Comment by jrose-apple** [comment]
Better than nothing. Thanks, Greg.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Pull request merged**